### PR TITLE
sbom workflow add

### PR DESCRIPTION
**Pull request by SadFaceSmith**
#### What this PR does
Quick PR to add SBOM report generation on release. We (Security Engineering) have this running in grafana/grafana and grafana/loki. We are planning to expand this to all our OSS Projects. 

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: .github/workflows/sbom-report.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    types: [created]
+
+jobs:
+  syft-sbom:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
```
Comment (aknuds1) [inline .github/workflows/sbom-report.yml:14]: Why not use v3?

File: CODEOWNERS [hunk 2]
```diff
@@ -12,5 +12,7 @@
 /Makefile @wilfriedroset @vaxvms @bubu11e @grafana/mimir-maintainers
 /packaging/ @wilfriedroset @vaxvms @bubu11e @grafana/mimir-maintainers
 
+# Security Workflows
+.github/workflows/sbom-report.yml @grafana/security-engineering
```
Comment (pstibrany) [inline CODEOWNERS:16]: Github complains about unknown owner here.

Comment (56quarters) [reply CODEOWNERS:16]: We also probably want `@grafana/mimir-maintainers` here.

Comment (SadFaceSmith) [reply CODEOWNERS:16]: Sounds good, let me know if any changes are needed 👍

Comment (56quarters) [reply CODEOWNERS:16]: > Sounds good, let me know if any changes are needed

@SadFaceSmith could you please remove this change to `CODEOWNERS`?

Comment (SadFaceSmith) [reply CODEOWNERS]: @56quarters Removed the `CODEOWNER` lines. 

@aknuds1 No, I'm not sure. I checked to make sure it's a real team in the `grafana` Org,

**Review by pracucci — APPROVED** [review]
Let's give it a try!

**Pull request merged**